neqo-common/src/qlog.rs

```diff
@@ -29,21 +30,53 @@ pub struct NeqoQlogShared {
 }
 
 impl NeqoQlog {
-    /// Create an enabled `NeqoQlog` configuration.
+    /// Create an enabled `NeqoQlog` configuration backed by a file.
     ///
     /// # Errors
     ///
-    /// Will return `qlog::Error` if cannot write to the new log.
-    pub fn enabled(
-        mut streamer: QlogStreamer,
-        qlog_path: impl AsRef<Path>,
+    /// Will return `qlog::Error` if it cannot write to the new file.
+    pub fn enabled_with_file(
+        mut qlog_path: PathBuf,
+        role: Role,
+        title: Option<String>,
+        description: Option<String>,
+        file_prefix: impl Display,
     ) -> Result<Self, qlog::Error> {
+        qlog_path.push(format!("{file_prefix}.sqlog"));
+
+        let file = OpenOptions::new()
+            .write(true)
+            // As a server, the original DCID is chosen by the client. Using
+            // create_new() prevents attackers from overwriting existing logs.
+            .create_new(true)
```

Comment:
Previously client would use `create`, thus potentially overriding an existing file. I can not think of a scenario where a client would want to override an existing qlog keyed by a cid. Thus I opted for using `create_new` for both client and server. Am I missing something?

https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.create

https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.create_new